File: in_Python/1281 Subtract the Product and Sum of Digits of an Integer.py
class Solution(object):
    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """
        res1 = 0
        res2 = 1
        while n != 0:
            res1 += n%10
            res2 *= n%10
            n /= 10

        return res2 - res1
        
        

File: in_Java/1281 Subtract the Product and Sum of Digits of an Integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        List<Integer> l = new ArrayList<>();
        while(n != 0){
            l.add(n % 10);
            n /= 10;
        }
        int a = 0;
        int b = 1;
        for(int i : l){
            a += i;
            b *= i;
        }
        return b - a;
    }
}